lib/apis/v3/node.go

```diff
@@ -98,6 +103,27 @@ type NodeWireguardSpec struct {
 	InterfaceIPv4Address string `json:"interfaceIPv4Address,omitempty" validate:"omitempty,ipv4"`
 }
 
+// NodeAddressType represents a node address type as a string
+type NodeAddressType string
+
+const (
+	NodeAddressIP                NodeAddressType = "IP"
+	NodeAddressVxlanTunnelIP     NodeAddressType = "VxlanTunnelIP"
```

Comment:
Nit: Go style convention is to put acronyms in upper case.  So, should be `NodeAddressVXLANTunnelIP`; k8s style convention is that all the characters in the acronym should be int he same case so should be `VXLANTunnelIP` (and we try to follow their guidelines for our v3 API).

Reply:
Reference: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions